src/index.js

```diff
@@ -601,6 +608,13 @@ export function useDropzone({
             })
             acceptedFiles.splice(0)
           }
+          if (multiple && maxNumber>1 &&  acceptedFiles.length > maxNumber) {
```

Comment:
```suggestion
          else if (multiple && maxFiles > 0 && acceptedFiles.length > maxFiles) {
```
But you can merge with the above condition and have a single block:
```js
if ((!multiple && acceptedFiles.length > 1) || (multiple && maxFiles > 0 && acceptedFiles.length > maxFiles)) {
  ...
```